src/main.cpp

```diff
@@ -23,20 +23,30 @@ struct GameState {
 	PhysicsObject player = PhysicsObject(vec3{0, map.get_height(0,0), 0});
 	Camera camera;
 	float camera_angle;
-	std::vector<Enemy> enemies;
+	double start_time = counter_in_seconds();
+	double next_enemy_time = counter_in_seconds();
 
 	GameState(const GameState &) = delete;
 
-	double next_enemy_time = counter_in_seconds() + ENEMY_DELAY;
 	void add_enemy_if_needed() {
 		if (counter_in_seconds() < this->next_enemy_time)
 			return;
 
-		log_printf("There are %d enemies, adding one more", (int)this->enemies.size());
 		float x, z;
 		Enemy::decide_location(this->player.get_location(), x, z);
-		this->enemies.push_back(Enemy(vec3{ x, this->map.get_height(x, z), z }));
-		this->next_enemy_time += ENEMY_DELAY;
+		this->map.add_enemy(Enemy(vec3{ x, this->map.get_height(x, z), z }));
+
+		/*
+		Later in the game, produce enemies more quickly.
+		DO NOT use something like "enemy_delay *= 0.99" because that will result in
+		an exponentially small enemy delay, i.e. too many enemies.
+		*/
+		double time_passed = counter_in_seconds() - this->start_time;
```

Comment:
Should make it more deterministic:

```suggestion
		double time_passed = this->next_enemy_time - this->start_time;
```